k8s/elastic-gke-logging/apptest/tester/tests/basic-suite.yaml

```diff
@@ -1,8 +1,217 @@
 actions:
-- name: dummy test
+- name: Wait until all expected nodes are connected to the Elasticsearch cluster
   bashTest:
     script: |-
-      exit 0
+      function get_nodes_count() {
+        local readonly health_url="$1"
+        nodes_count="$(curl -s "${health_url}" | jq '.number_of_nodes')"
+        echo "${nodes_count}"
+      }
+
+      expected_nodes="${REPLICAS}"
+
+      timeout=300
+      health_url="http://${APP_INSTANCE_NAME}-elasticsearch-svc:9200/_cluster/health"
+      timeout_time="$(( $(date +%s) + timeout ))"
+
+      # Wait for all nodes to join the cluster
+      nodes_count="$(get_nodes_count "${health_url}")"
+      while [[ "${nodes_count}" -ne "${expected_nodes}" ]] \
+          && [[ "$(date +%s)" -le "${timeout_time}" ]]; do
+        sleep 3
+        nodes_count="$(get_nodes_count "${health_url}")"
+      done
+
+      if [[ "${nodes_count}" -eq "${expected_nodes}" ]]; then
+        echo "OK - all nodes joined"
+      else
+        echo "FAIL - nodes count: ${nodes_count}"
+      fi
+    expect:
+      stdout:
+        contains: OK - all nodes joined
+
+- name: Wait for green status in the cluster
+  bashTest:
+    script: |-
+      function get_cluster_status() {
+        local readonly health_url="$1"
+        status="$(curl -s "${health_url}" | jq '.status')"
+        echo "${status}"
+      }
+
+      timeout=300
+      health_url="http://${APP_INSTANCE_NAME}-elasticsearch-svc:9200/_cluster/health"
+
+      timeout_time="$(( $(date +%s) + timeout ))"
+
+      # Wait for green status
+      status="$(get_cluster_status "${health_url}")"
+      while [[ "${status}" != '"green"' ]] \
+          && [[ "$(date +%s)" -le "${timeout_time}" ]]; do
+        sleep 3
+        status="$(get_cluster_status "${health_url}")"
+      done
+
+      if [[ "${status}" == '"green"' ]]; then
+        echo "OK - green status"
+      else
+        echo "FAIL - status: ${status}"
+      fi
+    expect:
+      stdout:
+        contains: OK - green status
+
+- name: Wait until GKE logs land in Elasticsearch
+  bashTest:
+    script: |-
+      elastic_url="http://${APP_INSTANCE_NAME}-elasticsearch-svc:9200"
+
+      timeout=300
+      timeout_time="$(( $(date +%s) + timeout ))"
+
+      # Wait for logstash indices being found
+      logstash_indices_count="$(curl -s "${elastic_url}/_cat/indices" | grep -c 'logstash-')"
+      while [[ "${logstash_indices_count}" -eq 0 ]] \
+          && [[ "$(date +%s)" -le "${timeout_time}" ]]; do
+        sleep 3
+        logstash_indices_count="$(curl -s "${elastic_url}/_cat/indices" | grep -c 'logstash-')"
+      done
+
+      if [[ "${logstash_indices_count}" -gt 0 ]]; then
+        echo "OK - logstash index found"
+      else
+        echo "FAIL - no logstash index"
+      fi
+    expect:
+      stdout:
+        contains: OK - logstash index found
+
+- name: Access Kibana UI
+  bashTest:
```

Comment:
Can you use `HttpTest` instead?

Reply:
As I understand, it would require modifying the tester.sh script to run envsubst against the test suite spec - I would prefer not do that. I can update it after the framework supports substituting the variables natively.